functions/util/ncUtils.js

```diff
@@ -41,7 +40,7 @@ class Stub {
         this.validatePayload();
 
         if (this.isValid) {
-            this.requestPromise = requestPromise;
+            this.requestPromise = require("request-promise");
```

Comment:
Since require() always returns the same object I suggest changing this to ```this.requestPromise = require("request-promise").defaults({});``` to avoid applying request-debug to the request-promise module. The affects of request-debug are seen in other channels which also require request-promise.

You may also want to move this line down under ```this.requestDefaults = ...``` and stick your defaults in there.

Reply:
I knew that request-debug was not being used exactly as I expected it to (was retaining some properties from whatever the first stub function to execute it on a give pod was).  That is why I moved the require down inside of the class constructor in hopes that would resolve the issue.  I was unaware that it was affecting other channels as well.

So you don't think that this change will prevent the issues.  Do you have a suggestion on how to apply request-debug only to this channel/function?  I was already planning to reach out if this didn't work because I am not understanding something obviously.

Regarding requestDefaults, request-debug does not play nice with the object returned by request.defaults(), which is why I do not call it here, but rather clone the defaults object for each request when made instead.

Reply:
So I've done some reading about how require caches the referenced module and why what I've done will not work.  I'm sure I knew this about require, but it never really sunk in until now.  I've read a little about how NOT to create a singleton and have some ideas.

Reply:
I'm not sure moving the require statement will make any difference because require is always going to return the same object once the module is fully loaded.

If request-debug doesn't work with the output of defaults() then I'm not sure there is any way to apply it to a specific channel/function.

Reply:
Right, I understand that now ([Node.js Docs](https://nodejs.org/api/modules.html#modules_caching) on require caching).

request-debug works with the output of defaults() using a hack that I just chose to avoid by not using defaults() (see code snippet).  But more importantly, I've tested it and using defaults() doesn't prevent the problem anyhow, due to how the mentioned hack gets at the underlying request module to do its work.
```javascript
  if (request.Request) {
    proto = request.Request.prototype
  } else if (request.get && request.post) {
    // The object returned by request.defaults() doesn't include the
    // Request property, so do this horrible thing to get at it.  Per
    // Wikipedia, port 4 is unassigned.
    var req = request('http://localhost:4').on('error', function() { })
    proto = req.constructor.prototype
  } else {
    throw new Error(
      "Pass the object returned by require('request') to this function.")
  }
```

We have 5 options I can think of:
1. remove request-debug and stop logging requests/responses for now.
2. implement a wrapper, similar to the one in the sdk to do the logging.
3. move the channel onto the sdk and implement that version of request.
4. bypass the require cache and load a fresh copy of request-promise every time.  This same thing is also done by request-promise internally to get a fresh copy of request.
5. Leave it alone, let it continue to be inaccurate and affect other channels as well.

I've implemented #4 using the "stealthy-require" module and updated the PR to show what that would look like.  Definitely open to discussion if we think that is a bad idea for some reason.